src/components/VideoRating.tsx

```diff
@@ -6,21 +6,35 @@ import { useState } from "react";
 interface VideoRatingProps {
   ratingProp?: number;
   votesCount?: number;
+  onGetNewRating: (rating: number) => void;
+  isAuthorized: boolean;
+  isRatingFreezen: boolean;
 }
-const VideoRating = ({ ratingProp, votesCount }: VideoRatingProps) => {
+const VideoRating = ({
+  ratingProp,
+  votesCount,
+  onGetNewRating,
+  isAuthorized,
+  isRatingFreezen,
+}: VideoRatingProps) => {
   const [rating, setRating] = useState(0);
-  const [hover] = useState(0);
-
+  const [hover]=useState(0);
+  
   return (
     <div className="star-rating">
       {[...Array(5)].map((star, index) => {
         const ratingUnit: number = index + 1;
+
         return (
           <button
+            style={isAuthorized && !isRatingFreezen? { cursor: "pointer" } : { cursor: "auto" }}
```

Comment:
style={{cursor: isRatingEnabled ? "pointer" : "default"}}

but it is better here use className for toggling cursor